compiler-core/src/ast.rs

```diff
@@ -1750,16 +1756,25 @@ impl TypedStatement {
     }
 
     pub fn find_node(&self, byte_index: u32) -> Option<Located<'_>> {
-        match self {
-            Statement::Use(_) => None,
-            Statement::Expression(expression) => expression.find_node(byte_index),
-            Statement::Assignment(assignment) => assignment.find_node(byte_index).or_else(|| {
-                if assignment.location.contains(byte_index) {
-                    Some(Located::Statement(self))
-                } else {
-                    None
+
+        // Search for the corresponding node inside the statement
+        // only if the index falls within the statement's location.
+        if self.location().contains(byte_index) {
```

Comment:
This is adding to the amount of work done rather than decreasing it. Calling `self.location()` pattern matches on self and then asks the inner value for the location, which is the same thing that the code below does.